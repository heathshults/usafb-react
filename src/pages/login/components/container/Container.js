import React from 'react';
import PropTypes from 'prop-types';

import backgroundImage from './background-image.png';
import './container.css';

const container = props => (
  <div className="container-fluid login__container" style={{ backgroundImage: `url(${backgroundImage})` }}>
    {React.Children.map(props.children, child => child)}
  </div>
);

container.propTypes = {
  children: PropTypes.array.isRequired,
};

export default container;
