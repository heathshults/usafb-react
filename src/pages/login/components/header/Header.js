import React from 'react';

const header = () => (
  <div className="col-md-8">
    <h1
      className="text-light ml-5"
      id="fadeInH1"
      style={{ display: 'none' }}
    >
      NATIONAL DATABASE
    </h1>
  </div>
);

export default header;
