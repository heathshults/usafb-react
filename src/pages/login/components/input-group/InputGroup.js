import React from 'react';
import PropTypes from 'prop-types';

import './inputGroup.css';

const inputGroup = props => (
  <div className="container">
    <form className="m-0 p-0" onSubmit={props.login}>
      {React.Children.map(props.children, child => child)}
    </form>
  </div>
);

inputGroup.propTypes = {
  login: PropTypes.func.isRequired,
  children: PropTypes.array.isRequired,
};

export default inputGroup;
