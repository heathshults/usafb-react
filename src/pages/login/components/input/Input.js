import React from 'react';
import PropTypes from 'prop-types';

import FontAwesome from 'react-fontawesome';

import './input.css';

const input = props => (
  <div className="row form-group">
    <div className="col-md-12 input-group">
      <label htmlFor="txtEmail" className="login-page__email-label sr-only">
        {props.placeholder}
      </label>
      <div className="col input-wrapper">
        <FontAwesome name={props.icon} className="fa-layer" />
        <input
          type={props.inputType}
          className="form-control form-control-theme"
          id="txtEmail"
          placeholder={props.placeholder}
          value={props.value}
          onChange={props.onChange}
        />
      </div>
    </div>
  </div>
);

input.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  icon: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  inputType: PropTypes.string
};

input.defaultProps = {
  inputType: 'text'
};

export default input;
