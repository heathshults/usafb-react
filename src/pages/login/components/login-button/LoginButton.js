import React from 'react';
import PropTypes from 'prop-types';

import './login-button.css';

const getButtonLabel = (loggingIn) => {
  if (loggingIn) {
    return (
      <div>
        <i className="fa fa-spinner fa-spin" /> Logging in...
      </div>
    );
  }

  return (
    <div>
      Login
    </div>
  );
};

const loginButton = props => (
  <div className="row pt-0 pb-0">
    <div className="col-md-12 pt-1 pb- text-center bss">
      <button
        type="submit"
        className="btn-blue-grad-pill login__btn w-75 w-50 mt-2 mx-auto"
        disabled={props.loggingIn}
      >
        {getButtonLabel(props.loggingIn)}
      </button>
    </div>
  </div>
);

loginButton.propTypes = {
  loggingIn: PropTypes.bool.isRequired
};

export default loginButton;
