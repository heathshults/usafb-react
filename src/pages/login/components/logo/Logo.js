import React from 'react';

import logoImage from 'images/logo/usafb-logo.png';
import './logo.css';

const logo = () => (
  <img src={logoImage} alt="Blue Star Sports" className="login__logo img-fluid" />
);

export default logo;
