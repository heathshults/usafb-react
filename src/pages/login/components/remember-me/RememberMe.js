import React from 'react';
import PropTypes from 'prop-types';

import './remember-me.css';

const rememberMe = props => (
  <div className="row form-group d-inline-flex mt-0 pt-0">
    <div className="col-md-12 input-group justify-content-center mt-0 pt-0">
      <a
        className="text-white remember-me__forgot-button"
        role="button"
        onClick={props.forgotPassword}
        tabIndex={0}
      >
        <small>
          Forgot Password?
        </small>
      </a>
    </div>
  </div >
);

rememberMe.propTypes = {
  forgotPassword: PropTypes.func.isRequired
};

export default rememberMe;
