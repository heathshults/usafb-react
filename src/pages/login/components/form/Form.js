import React from 'react';
import PropTypes from 'prop-types';

import Logo from '../logo/Logo';

import './form.css';

const form = props => (
  <div className="col-md-4 text-center">
    <Logo />
    <div id="si-form" className="width-si-form">
      <h1>NATIONAL DATABASE</h1>
      {React.Children.map(props.children, child => child)}
    </div>
  </div>
);

form.propTypes = {
  children: PropTypes.array.isRequired
};

export default form;
