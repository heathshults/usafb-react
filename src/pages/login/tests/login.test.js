import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
// import configureStore from 'redux-mock-store';

import LoginWithRedux, { Login } from '../Login';
import Container from '../components/container/Container';
import Header from '../components/header/Header';
import Form from '../components/form/Form';
import Logo from '../components/logo/Logo';
import ErrorMessage from '../components/error/Error';
import InputGroup from '../components/input-group/InputGroup';
import Input from '../components/input/Input';
import RememberMe from '../components/remember-me/RememberMe';
import LoginButton from '../components/login-button/LoginButton';

const setupLoginComponent = () => {
  const props = {
    loginReducer: {
      loginError: '',
      loggingIn: false
    },
    login: () => { }
  };

  const loginWrapper = mount(<Login {...props} />);

  return {
    props,
    loginWrapper
  };
}

describe('[LOGIN PAGE] UI components test', () => {
  const { loginWrapper } = setupLoginComponent();

  test('container exists', () => {
    expect(loginWrapper.find(Container).length).toBe(1);
  });

  test('header exists', () => {
    expect(loginWrapper.find(Header).length).toBe(1);
  });

  test('form exists', () => {
    expect(loginWrapper.find(Form).length).toBe(1);
  });

  test('logo exists', () => {
    expect(loginWrapper.find(Logo).length).toBe(1);
  });

  test('error message exists', () => {
    expect(loginWrapper.find(ErrorMessage).length).toBe(1);
  });

  test('input group exists', () => {
    expect(loginWrapper.find(InputGroup).length).toBe(1);
  });

  test('email and password text fields exists', () => {
    expect(loginWrapper.find(Input).length).toBe(2);
  });

  test('remember me exists', () => {
    expect(loginWrapper.find(RememberMe).length).toBe(1);
  });

  test('login button exists', () => {
    expect(loginWrapper.find(LoginButton).length).toBe(1);
  });
});

describe('[LOGIN PAGE] functionality test', () => {
  const { loginWrapper } = setupLoginComponent();

  test('Login functionality', () => {

    // console.log(process.env);
    // get a snapshot of the login screen without any user input
    let component = renderer.create(loginWrapper);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    loginWrapper.setState({
      email: 'footballautomation@gmail.com',
      password: 'password123'
    });

    loginWrapper.find(Input).first().value = loginWrapper.state().email;
    loginWrapper.find(Input).last().value = loginWrapper.state().password;

    // get a snapshot of user inputs
    component = renderer.create(loginWrapper);
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    const loginButton = loginWrapper.find(LoginButton).simulate('click');
    const loggedInComponent = renderer.create(loginWrapper);
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});