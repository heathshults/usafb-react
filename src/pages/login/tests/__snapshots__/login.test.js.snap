// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[LOGIN PAGE] functionality test Login functionality 1`] = `
<div
  className="container-fluid login__container"
  style={
    Object {
      "backgroundImage": "url(background-image.png)",
    }
  }
>
  <div
    className="row align-items-center w-100"
  >
    <div
      className="col-md-8"
    >
      <h1
        className="text-light ml-5"
        id="fadeInH1"
        style={
          Object {
            "display": "none",
          }
        }
      >
        NATIONAL DATABASE
      </h1>
    </div>
    <div
      className="col-md-4"
    >
      <div
        className="width-si-form"
        id="si-form"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <img
          alt="Blue Star Sports"
          className="login__logo img-fluid"
          src="usafb-logo.png"
        />
        <p
          className="login__error-message"
        >
          
        </p>
        <div
          className="container"
        >
          <form
            className="input-group__form-container"
            onSubmit={[Function]}
          >
            <div
              className="row form-group"
            >
              <div
                className="col-md-12 input-group"
              >
                <label
                  className="login-page__input-label sr-only"
                  htmlFor="txtEmail"
                >
                  Email
                </label>
                <div
                  className="col input-wrapper"
                >
                  <span
                    aria-hidden={true}
                    className="fa fa-envelope fa-layer"
                  />
                  <input
                    className="form-control form-control-theme"
                    id="txtEmail"
                    onChange={[Function]}
                    placeholder="Email"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              className="row form-group"
            >
              <div
                className="col-md-12 input-group"
              >
                <label
                  className="login-page__input-label sr-only"
                  htmlFor="txtEmail"
                >
                  Password
                </label>
                <div
                  className="col input-wrapper"
                >
                  <span
                    aria-hidden={true}
                    className="fa fa-lock fa-layer"
                  />
                  <input
                    className="form-control form-control-theme"
                    id="txtEmail"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              className="row form-group d-inline-flex"
            >
              <div
                className="col-md-12 input-group justify-content-center"
              >
                <label
                  className="form-check-label"
                  htmlFor="rememberMe"
                >
                  <span
                    className="checkbox-theme"
                  >
                    <input
                      className="form-check-input text-center"
                      id="rememberMe"
                      type="checkbox"
                    />
                    <small
                      className="text-white"
                    >
                      Remember Me
                    </small>
                  </span>
                </label>
                <a
                  className="text-white ml-5"
                >
                  <small>
                    Forgot Password?
                  </small>
                </a>
              </div>
            </div>
            <div
              className="row pt-0 pb-0"
            >
              <div
                className="col-md-12 pt-0 pb- text-center"
              >
                <button
                  className="btn btn-primary w-75 mr-auto ml-auto"
                  type="submit"
                >
                  Login
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`[LOGIN PAGE] functionality test Login functionality 2`] = `
<div
  className="container-fluid login__container"
  style={
    Object {
      "backgroundImage": "url(background-image.png)",
    }
  }
>
  <div
    className="row align-items-center w-100"
  >
    <div
      className="col-md-8"
    >
      <h1
        className="text-light ml-5"
        id="fadeInH1"
        style={
          Object {
            "display": "none",
          }
        }
      >
        NATIONAL DATABASE
      </h1>
    </div>
    <div
      className="col-md-4"
    >
      <div
        className="width-si-form"
        id="si-form"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <img
          alt="Blue Star Sports"
          className="login__logo img-fluid"
          src="usafb-logo.png"
        />
        <p
          className="login__error-message"
        >
          
        </p>
        <div
          className="container"
        >
          <form
            className="input-group__form-container"
            onSubmit={[Function]}
          >
            <div
              className="row form-group"
            >
              <div
                className="col-md-12 input-group"
              >
                <label
                  className="login-page__input-label sr-only"
                  htmlFor="txtEmail"
                >
                  Email
                </label>
                <div
                  className="col input-wrapper"
                >
                  <span
                    aria-hidden={true}
                    className="fa fa-envelope fa-layer"
                  />
                  <input
                    className="form-control form-control-theme"
                    id="txtEmail"
                    onChange={[Function]}
                    placeholder="Email"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              className="row form-group"
            >
              <div
                className="col-md-12 input-group"
              >
                <label
                  className="login-page__input-label sr-only"
                  htmlFor="txtEmail"
                >
                  Password
                </label>
                <div
                  className="col input-wrapper"
                >
                  <span
                    aria-hidden={true}
                    className="fa fa-lock fa-layer"
                  />
                  <input
                    className="form-control form-control-theme"
                    id="txtEmail"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              className="row form-group d-inline-flex"
            >
              <div
                className="col-md-12 input-group justify-content-center"
              >
                <label
                  className="form-check-label"
                  htmlFor="rememberMe"
                >
                  <span
                    className="checkbox-theme"
                  >
                    <input
                      className="form-check-input text-center"
                      id="rememberMe"
                      type="checkbox"
                    />
                    <small
                      className="text-white"
                    >
                      Remember Me
                    </small>
                  </span>
                </label>
                <a
                  className="text-white ml-5"
                >
                  <small>
                    Forgot Password?
                  </small>
                </a>
              </div>
            </div>
            <div
              className="row pt-0 pb-0"
            >
              <div
                className="col-md-12 pt-0 pb- text-center"
              >
                <button
                  className="btn btn-primary w-75 mr-auto ml-auto"
                  type="submit"
                >
                  Login
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`[LOGIN PAGE] functionality test Login functionality 3`] = `
<div
  className="container-fluid login__container"
  style={
    Object {
      "backgroundImage": "url(background-image.png)",
    }
  }
>
  <div
    className="row align-items-center w-100"
  >
    <div
      className="col-md-8"
    >
      <h1
        className="text-light ml-5"
        id="fadeInH1"
        style={
          Object {
            "display": "none",
          }
        }
      >
        NATIONAL DATABASE
      </h1>
    </div>
    <div
      className="col-md-4"
    >
      <div
        className="width-si-form"
        id="si-form"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <img
          alt="Blue Star Sports"
          className="login__logo img-fluid"
          src="usafb-logo.png"
        />
        <p
          className="login__error-message"
        >
          
        </p>
        <div
          className="container"
        >
          <form
            className="input-group__form-container"
            onSubmit={[Function]}
          >
            <div
              className="row form-group"
            >
              <div
                className="col-md-12 input-group"
              >
                <label
                  className="login-page__input-label sr-only"
                  htmlFor="txtEmail"
                >
                  Email
                </label>
                <div
                  className="col input-wrapper"
                >
                  <span
                    aria-hidden={true}
                    className="fa fa-envelope fa-layer"
                  />
                  <input
                    className="form-control form-control-theme"
                    id="txtEmail"
                    onChange={[Function]}
                    placeholder="Email"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              className="row form-group"
            >
              <div
                className="col-md-12 input-group"
              >
                <label
                  className="login-page__input-label sr-only"
                  htmlFor="txtEmail"
                >
                  Password
                </label>
                <div
                  className="col input-wrapper"
                >
                  <span
                    aria-hidden={true}
                    className="fa fa-lock fa-layer"
                  />
                  <input
                    className="form-control form-control-theme"
                    id="txtEmail"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              className="row form-group d-inline-flex"
            >
              <div
                className="col-md-12 input-group justify-content-center"
              >
                <label
                  className="form-check-label"
                  htmlFor="rememberMe"
                >
                  <span
                    className="checkbox-theme"
                  >
                    <input
                      className="form-check-input text-center"
                      id="rememberMe"
                      type="checkbox"
                    />
                    <small
                      className="text-white"
                    >
                      Remember Me
                    </small>
                  </span>
                </label>
                <a
                  className="text-white ml-5"
                >
                  <small>
                    Forgot Password?
                  </small>
                </a>
              </div>
            </div>
            <div
              className="row pt-0 pb-0"
            >
              <div
                className="col-md-12 pt-0 pb- text-center"
              >
                <button
                  className="btn btn-primary w-75 mr-auto ml-auto"
                  type="submit"
                >
                  Login
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
